nbtc/relayer_test.go

```diff
@@ -71,46 +74,104 @@ func TestNewRelayer_ErrorCases(t *testing.T) {
 	btcProcessor, _ := ika2btc.NewProcessor(btcClientConfig, 6, db)
 	btcProcessor.BtcClient = &bitcoin.MockClient{}
 	nativeProcessor := native2ika.NewProcessor(mockIkaClient, db)
+	mockFetcher, _ := native2ika.NewMockAPISignRequestFetcher()
 
 	testCases := []struct {
 		name            string
 		db              *dal.DB
 		nativeProcessor *native2ika.Processor
 		btcProcessor    *ika2btc.Processor
 		expectedError   error
+		fetcher         native2ika.SignReqFetcher
 	}{
 		{
 			name:            "DatabaseError",
 			db:              nil,
 			nativeProcessor: nativeProcessor,
 			btcProcessor:    btcProcessor,
 			expectedError:   err.ErrNoDB,
+			fetcher:         mockFetcher,
 		},
 		{
 			name:            "NativeProcessorError",
 			db:              db,
 			nativeProcessor: nil,
 			btcProcessor:    btcProcessor,
 			expectedError:   err.ErrNoNativeProcessor,
+			fetcher:         mockFetcher,
 		},
 		{
 			name:            "BtcProcessorError",
 			db:              db,
 			nativeProcessor: nativeProcessor,
 			btcProcessor:    nil,
 			expectedError:   err.ErrNoBtcProcessor,
+			fetcher:         mockFetcher,
+		},
+		{
+			name:            "BlockchainError",
+			db:              db,
+			nativeProcessor: nativeProcessor,
+			btcProcessor:    btcProcessor,
+			expectedError:   err.ErrNoFetcher,
+			fetcher:         nil,
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			relayer, err := NewRelayer(relayerConfig, tc.db, tc.nativeProcessor, tc.btcProcessor)
+			relayer, err := NewRelayer(relayerConfig, tc.db, tc.nativeProcessor, tc.btcProcessor, tc.fetcher)
 			assert.ErrorIs(t, err, tc.expectedError)
 			assert.Assert(t, relayer == nil)
 		})
 	}
 }
 
+func TestRelayer_fetchAndStoreNativeSignRequests(t *testing.T) {
+	db := initTestDB(t)
+	mockIkaClient := ika.NewMockClient()
+	btcProcessor, _ := ika2btc.NewProcessor(btcClientConfig, 6, db)
+	btcProcessor.BtcClient = &bitcoin.MockClient{}
+	nativeProcessor := native2ika.NewProcessor(mockIkaClient, db)
+	mockFetcher, err := native2ika.NewMockAPISignRequestFetcher()
+	assert.NilError(t, err)
+
+	relayer, err := NewRelayer(relayerConfig, db, nativeProcessor, btcProcessor, mockFetcher)
+	assert.NilError(t, err)
+
+	err = relayer.fetchAndStoreNativeSignRequests()
+	assert.NilError(t, err)
+
+	assert.Equal(t, relayer.signReqStart, 5) // Should be 5 after fetching 5 sign requests
+
+	requests, err := db.GetPendingIkaSignRequests()
+	assert.NilError(t, err)
+	assert.Equal(t, len(requests), 5) // Should be 5 inserted requests
+}
+
+func TestRelayer_storeSignRequest(t *testing.T) {
+	db := initTestDB(t)
```

Comment:
Let's create test suite to avoid duplicate code.

Reply:
You can see TestRelayer_fetchAndStoreNativeSignRequests and TestRelayer_storeSignRequest share the same code to create context for testing.

Reply:
Done